Microsoft.Toolkit.Uwp.UI.Controls/Menu/Menu.Events.cs

```diff
@@ -198,16 +215,67 @@ private void Menu_Loaded(object sender, RoutedEventArgs e)
                 _wrapPanel.Orientation = Orientation;
             }
 
+            Window.Current.CoreWindow.PointerMoved -= CoreWindow_PointerMoved;
             LostFocus -= Menu_LostFocus;
             LostFocus += Menu_LostFocus;
             Dispatcher.AcceleratorKeyActivated -= Dispatcher_AcceleratorKeyActivated;
             Window.Current.CoreWindow.KeyDown -= CoreWindow_KeyDown;
             Dispatcher.AcceleratorKeyActivated += Dispatcher_AcceleratorKeyActivated;
             Window.Current.CoreWindow.KeyDown += CoreWindow_KeyDown;
+            Window.Current.CoreWindow.PointerMoved += CoreWindow_PointerMoved;
+
+            var ttv = TransformToVisual(Window.Current.Content);
+            Point screenCoords = ttv.TransformPoint(new Point(0, 0));
+            _x1 = screenCoords.X;
+            _y1 = screenCoords.Y;
+            _x2 = _x1 + ActualWidth;
+            _y2 = _y1 + ActualHeight;
+        }
+
+        private void CoreWindow_PointerMoved(CoreWindow sender, PointerEventArgs args)
+        {
+            // if contained with the whole Menu control
+            if (IsOpened && WithinRange(_x1, _x2, _y1, _y2, args.CurrentPoint.Position.X, args.CurrentPoint.Position.Y))
+            {
+                // if hover over current opened item
+                if (WithinRange(
+                    SelectedMenuItem.X1,
+                    SelectedMenuItem.X2,
+                    SelectedMenuItem.Y1,
+                    SelectedMenuItem.Y2,
+                    args.CurrentPoint.Position.X,
+                    args.CurrentPoint.Position.Y))
+                {
+                    return;
+                }
+
+                // TODO to be replaced with Range tree or any faster datastructure
+                    foreach (MenuItem menuItem in Items)
+                {
+                    if (WithinRange(
+                        menuItem.X1,
+                        menuItem.X2,
+                        menuItem.Y1,
+                        menuItem.Y2,
+                        args.CurrentPoint.Position.X,
+                        args.CurrentPoint.Position.Y))
+                    {
+                        if (menuItem == SelectedMenuItem)
```

Comment:
The check inside the loop is not required as it shouldn't be possible to reach the loop if SelectedMenuItem is the current item but I added just in case.
Will remove the check from inside the loop